Tests/Datagrid/ProxyQueryTest.php

```diff
@@ -99,10 +100,19 @@ public function testGetFixedQueryBuilder($class, $alias, $id, $expectedId, $valu
         $qb->expects($this->once())
             ->method('setParameter')
             ->with($this->equalTo($expectedId), $this->equalTo(array($value)));
-        $qb->expects($this->once())
+        $qb->expects($this->any())
             ->method('getDQLPart')
-            ->with($this->equalTo('from'))
-            ->willReturn(array(new From($class, $alias)));
+            ->will($this->returnCallBack(function ($part) use ($class, $alias) {
+                $parts = array(
+                    'from' => array(new From($class, $alias)),
+                    'orderBy' => array(new OrderBy('whatever', 'DESC')),
+                );
+
+                return $parts[$part];
+            }));
+        $qb->expects($this->once())
+            ->method('addOrderBy')
+            ->with("$alias.$id", null);
```

Comment:
Thanks for being understanding, I was like you before, because I "heard" if was bad, then lately I realized it is bad in some cases, and in some very simple other cases, it's better because it is way more readable. It's a bit like Yoda conditions if you will : a safe-guard that protects you from errors you don't make anymore, but bit you hard one time and made you decide to embrace it. But now that we have code review, I think we can avoid these mistakes easily.